projects/core/src/auth/user-auth/config/default-auth-config.ts

```diff
@@ -4,10 +4,41 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import { inject } from '@angular/core';
-import { FeatureToggles } from '../../../features-config';
+import { inject, InjectionToken, ValueProvider } from '@angular/core';
 import { AuthConfig } from './auth-config';
 
+const USE_AUTHORIZATION_CODE_FLOW_BY_DEFAULT = new InjectionToken<boolean>(
+  'USE_AUTHORIZATION_CODE_FLOW_BY_DEFAULT',
+  {
+    factory: () => false,
+  }
+);
+
+/**
+ * When enabled, sets the default oAuth configuration to use authorization
+ * code flow with PKCE. This results in a more secure authorization scheme
+ * as the default configuration.
+ *
+ * NOTE: This flag should only be enabled when used with a CCv2 Authorization
+ * Server running the September 2025 update or higher. The CCv2 Authorization
+ * Server only supports Authorization Code flow for public clients from
+ * that version and onwards.
+ *
+ * @usageNotes
+ * Add to the root module providers:
+ * ```
+ * useAuthorizationCodeFlowByDefaultProvider(true)
+ * ```
+ */
+export function useAuthorizationCodeFlowByDefaultProvider(
```

Comment:
Angular API (and we try to mimic it) uses by convention the word `provide` as the first word in such providers utils
Angular: e.g. `provideClientHydration()`, `provideRoutes()`, ...
Spartacus: e.g. `provideConfig()`, `provideFeatureToggles()`, ...

What about `provideAuthorizationCodeFlowByDefault()` ?